### ISSUE TITLE
xkill won't launch from click-commands or menu-x-x-exec commands. 

### ISSUE THREAD

**Issue by cyberrumor** [jaagr/polybar#1702]
## Describe the issue
I have "menu-1-1-exec = xkill" in the only "type = custom/menu" module in my bar, but clicking the associated menu button causes the menu to close without launching xorg-xkill. I have also tried to launch xkill via "click-left = xkill" in a "type = custom/text" module, and can't get it to launch like that either. Typing xkill into urxvt or dmenu causes xkill to launch normally. I have also tried tying ">/dev/null 2>&1" and "& disown" in every combination to the end of these commands, as well as attempting to use the full path to the command of "/bin/xkill" with no success. I have also tried "urxvt -e xkill" since it works when typed into urxvt normally, also without success. 

**Expected behavior:**

Expected behavior is that when the button is clicked, cursor turns into an x, and the next window that is clicked on has its connection to the x server cut off, which typically causes the window's pid(s) to be killed as well. 

**Actual behavior:**

Actual behavior is that the xkill-associated button causes the menu to collapse, but xkill does not initiate. I've tried running polybar with every log level to see if logging produced an error as to why, with no success. 

**Was it working before?**
I have never had xkill successfully launch, but this is also the first time I've tried. 

* Did you also experience this bug in an earlier version of polybar?
No, currently running version 3.3.0 [Features: +alsa +curl +i3 -mpd +network(libnl) +pulseaudio +xkeyboard].

* If no [sic], what was the last version where this worked correctly?
N/A

## To Reproduce
*A minimal but complete config with which the problem occurs:* Am using colors generated by pywal. 
```
[settings]
pseudo-transparency = false

[colors]
background = ${xrdb:color0:#222}
foreground = ${xrdb:color7:#222}
foreground-alt = ${xrdb:color6:#222}
primary = ${xrdb:color1:#222}
secondary = ${xrdb:color2:#222}
alert = ${xrdb:color3:#222}

[bar/mybar]
modules-left = linuxmenu
modules-center = xkill
modules-right = 
background = ${colors.background}
foreground = ${colors.foreground}
width = 100%
height = 20
offset-x = 0
offset-y = 0
override-redirect = false
tray-position = right
font-0 = "overpass-regular:size=12;1:antialias=true:hinting=true"
font-1 = "MaterialIcons-Regular:size=12;1:antialias=true:hinting=true"
padding = 1

[module/linuxmenu]
type = custom/menu
expand-right=true
label-open = menu
label-open-padding = 4
label-close = menu
label-close-padding = 4

menu-0-0 = menutowardxkill
menu-0-0-exec = menu-open-1
menu-0-1 = powermenu    |
menu-0-1-exec = menu-open-2

menu-1-0 = update
menu-1-0-exec = urxvt -name foo -e sudo pacman -Syu
menu-1-1 = shouldLaunchXkill
menu-1-1-exec = xkill
menu-1-2 = i3 settings via notepadqq
menu-1-2-exec = notepadqq /home/$USER/.config/i3/config & disown
menu-1-3 = polybar settings via notepadqq
menu-1-3-exec = notepadqq /home/$USER/.config/polybar/config & disown
;using literal username instead of EV in actual config, not sure if it matters.
menu-1-4 = clean orphans    |
menu-1-4-exec = urxvt -name foo -e sudo pacman -Rns $(pacman -Qtdq)

menu-2-0 = shutsdownRN
menu-2-0-exec = shutdown -h now
menu-2-1 = rebootsRN
menu-2-1-exec = reboot
menu-2-2 = LocksScreenWithI3Lock-next  |
menu-2-2-exec = i3lock-next "password" "Overpass-Regular" 24

label-separator = |
label-separator-padding = 4

[module/xkill]
type = custom/text
content = xkillicon
content-padding = 4
click-left = xkill &

```
*List any other steps needed to reproduce the issue besides starting polybar with the config you posted above.* 

requires xorg-xkill. I am using Arch Linux x86_64 with kernel 5.0.2-arch1-1-ARCH, according to neofetch. Shell: bash 5.0.2. Xkill version is 1.0.5. Overpass is the font used. 

## Polybar Log
*Post everything polybar outputs to the terminal when you run it and the issue occurs below*
```
[cyberrumor@Archmage /]$ polybar -l trace mybar

-Found matching input area
*Uncaught input event, forwarding to shell... (input: xkill)
* Executing shell command: xkill
- command: Waiting for pid 4842 to finish...
-command: Exited with failed status 1
*module/linuxmenu: Rebuilding cache


```

## Screenshots
*If applicable, add screenshots to help explain your problem.*
NA: would require a video. 

## Environment:
* WM: i3-gaps version 4.16.1 (2019-01-27) @ 2009 Michael Stapelberg and contributors
* Distro: Arch 
* Output of `polybar -vvv`:

polybar 3.3.0 
Features: +alsa +curl +i3 -mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +xrandr (+monitors) -render -damage -sync +composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -02 -pipe -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -pedantic -pedantic-errors
Linker flags: -Wl , -01 , --sort-common , --as-needed , -z , relro , -z , now

## Additional context
*Add any other context that you think is necessary about the problem here.*
I'm sorry if you had to read all this and it's an issue with my config or with like xkill forking or something like that. Thanks for the brilliant program :)

**Comment by Lomadriel** [comment]
If the output is not redirected the program will crash if you use `&` at the end of the command, this is a know bug (#916). See also https://github.com/jaagr/polybar/wiki/Known-Issues#click-commands-are-not-executed

I tried something in the branch of one of my PR #1680 that fixes #916. xkill is launched as intended however it still not work because `xkill` cannot be launched when the cursor is pressed (see log below). You can try it by pressing your left button while hitting the enter key in your terminal.

As a workaround I tried to bind `click-right` but in fact `xkill` can't be launched when any mouse button is pressed :disappointed:. However `click-left = sleep 0.1 && xkill > /dev/null 2>&1 &` works. This is not the best but for now polybar doesn't support launching command when the mouse button is released. (This could be a feature idea though)

```
[…]
* Uncaught input event, forwarding to shell... (input: xkill &)
* Executing shell command: xkill &
- command: Waiting for pid 4419 to finish...
- command: Exited with status 0
- bar: Force update
* Redrawing bar window
- renderer: flush(1 geom=91x20+0+0, falloff=0)
- renderer: flush(2 geom=103x20+909+0, falloff=0)
Select the window whose client you wish to kill with button 1....
xkill:  unable to grab cursor
```

**Comment by cyberrumor** [comment]
Perfect. Honestly, a workaround is good enough for me. I can't believe I didn't think to try that. Closing. Thanks for your help, Lomadriel :)
